src/main/java/com/carmen/app/services/CarService.java

```diff
@@ -30,13 +38,14 @@ public void createCar(Car car) {
 	
 	}
 	
-	public void updateCar(UUID id) {
-	em.find(Car.class, id);
+	public void updateCar(Car car) {
+		getCar(car.getId());
 		em.merge(car);
```

Comment:
in the merge you must inform the entity returned from the getCar method.
What if the getCar returns 'null' ?